### DISCUSSION TITLE
Alignment read counts by sample when using input file name expansion

### DISCUSSION THREAD

**Discussion by benjytan88** [milaboratory/mixcr#1784] [Q&A] [answered]
Hi MiXcR team, 
I'm using MiXcR to analyze multiple samples of BCR data, so I am using file name expansion for it. My FASTQ files look like this with the sample name at the front.

```
LN1_R1_001.fastq.gz
LN1_R2_001.fastq.gz
LN2_R1_001.fastq.gz
LN2_R2_001.fastq.gz
SC1_R1_001.fastq.gz
SC1_R2_001.fastq.gz
SC2_R1_001.fastq.gz
SC2_R2_001.fastq.gz
```

My command to run MiXcR is as follows: 
`mixcr analyze --output-not-used-reads --split-by-sample neb-mouse-rna-xcr-umi-nebnext ./FASTQs/{{a}}_{{R}}_001.fastq.gz ./OUTPUT/`

Upon completion, when I checked the QC, all of them have the same number of reads sequenced and % successfully aligned reads. When I generate the alignment reports using this command, all my barplots have the same distribution. 

`mixcr exportQc align ./OUTPUT/*.vdjca ./REPORT/alignQC.pdf`

 I believe this shows the total sequences aligned for all samples together, but I wanted is the QC for each individual sample. Could you please guide on how can I get this? Am I missing any parameters in my command? 

Thank you.

**Comment by mizraelson** [comment, marked as answer]
Hi,
When you use `{{a}}_{{R}}_001.fastq.gz`, it merges everything together and processes it as a single sample.
If you want to process the samples separately, you can use:
`{{SMPL:a}}_{{R}}_001.fastq.gz`.

**Comment by benjytan88** [comment]
Hi @mizraelson , thanks for your reply. I realized my error there and adjusted my command line to as follows: 
`mixcr analyze --output-not-used-reads --split-by-sample neb-mouse-rna-xcr-umi-nebnext ./FASTQs/{{SAMPLE:a}}_{{R}}_001.fastq.gz ./OUTPUT/`

However, I still get the same problem. Here's a snapshot of the alignQC report I got from the latest run. 

![Res_11Sept](https://github.com/user-attachments/assets/473978ef-7a0e-434e-8489-884d18b9530c)

What else can I do to correct this?

**Reply by mizraelson** [reply]
When you use file name expansion, the samples are aligned together and are only separated afterward. This is why you receive a joint align report for them. All other reports are separated by sample.
If you want to align samples separately, the only way is to run the command for each sample individually. To make this process easier, I recommend using [GNU parallel](https://www.gnu.org/software/parallel/). The command will look like the one below:

```
ls ./FASTQs/*R1_001.fastq.gz | parallel -j1 "mixcr analyze neb-mouse-rna-xcr-umi-nebnext \
    --output-not-used-reads \
    {} {=s:R1:R2:=} \
    {=s:.*/OUTPUT/:;s:_R.*::=}"
```

**Reply by benjytan88** [reply]
I see. Thanks for the information. Will test it out.
